### PR TITLE
Allow defining a fixed port for datasource DevServices

### LINKED ISSUE #16768: Being able to fix the Postgres listening port property with DevServices
## Description

When you use DevServices in Dev and Test mode to startup a Postgres database, the listening port is random and you get a different one each time you startup or execute the tests (eg. `localhost:55046/default`, `localhost:55048/default`, etc.). But in development mode, it's sometimes useful to access the database. And to get the port, you need to check the Dev UI to know this random port.

Having a random port in test, is fine, but in dev it would be nice to have to fix the port number (using a property).

* Zulip Discussion : https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Fix.20the.20Postgres.20listening.20port.20property.20with.20DevServices

### PR DESCRIPTION
**Pull request by gsmet**
Fixes #16768

@agoncal could you check if this satisfies your requirements? I could be convinced to backport it to the next 1.13 that I will release next Wednesday.

### CODE REVIEW

File: extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DevServicesBuildTimeConfig.java [hunk 1]
```diff
@@ -32,4 +33,12 @@
...
     @ConfigItem
     public Map<String, String> properties;
+
+    /**
+     * Optional fixed port the dev service will listen to.
+     * <p>
+     * If not defined, the port will be chosen randomly.
+     */
+    @ConfigItem
+    public OptionalInt port;
```
Comment (machi1990) [inline extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DevServicesBuildTimeConfig.java:43]: Are we planning to add the same capability for Mongo dev services https://github.com/quarkusio/quarkus/blob/2cd3b3b4a7d489614bf1e3e8e46f359af5d94723/extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/DevServicesBuildTimeConfig.java#L9?

Comment (gsmet) [reply extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DevServicesBuildTimeConfig.java:43]: I added a commit with it.

Comment (machi1990) [reply extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DevServicesBuildTimeConfig.java:43]: Awesome, thanks!

**Review by machi1990 — APPROVED** [review]

**Review by gastaldi — APPROVED** [review]
LGTM

**Pull request merged**